main.c

```diff
@@ -74,6 +30,7 @@ int main(int argc, char** argv){
     // At least one argument, and not more than the max amount of files
     if (argc == 1){
         printf("No file provided!\n");
+        help();
```

Comment:
This should probably only print help in the for loop below.

Reply:
Actually you want both, but to different locations (stdout vs stderr).